src/stateful-event-subscriber.ts

```diff
@@ -255,8 +256,9 @@ export abstract class StatefulEventSubscriber<State>
       }
       //Find the last state before the blockNumber of the logs
       let stateBeforeLog: DeepReadonly<State> | undefined;
-      for (const bn in this.stateHistory) {
-        if (+bn >= blockNumber) break;
+      for (const _bn of Object.keys(this.stateHistory)) {
+        const bn = +_bn;
```

Comment:
this intermediary step is not necessary because
`object[keyAsNum] === object[String(keyAsNum)]` is truthy